tests/rustdoc-json/reexport/issue-112852-dangling-trait-impl-id.rs

```diff
@@ -0,0 +1,12 @@
+#![feature(no_core)]
+#![no_core]
+
+// @has "$.index[*][?(@.name=='PubTrait')]"
+pub trait PubTrait {}
+
```

Comment:
It's worth being more explicit about what we're testing


```suggestion
// @!has "$.index[*][?(@.name == 'HiddenPubStruct')]"
// @!has "$.index[*][?(@.inner.impl)]"
```